db/types/base.py

```diff
@@ -66,25 +137,82 @@ class PostgresType(Enum):
     UUID = 'uuid'
 
 
-class MathesarCustomType(Enum):
+# Since we want to have our identifiers quoted appropriately for use in
+# PostgreSQL, we want to use the postgres dialect preparer to set this up.
+preparer = sa_create_engine("postgresql://").dialect.identifier_preparer
+
+
+SCHEMA = f"{constants.MATHESAR_PREFIX}types"
+
+# Should usually equal `mathesar_types`
+_ma_type_qualifier_prefix = preparer.quote_schema(SCHEMA)
+
+
+# TODO rename to get_qualified_mathesar_obj_name
+# it's not only used for types. it's also used for qualifying sql function ids
+def get_qualified_name(unqualified_name):
+    return ".".join([_ma_type_qualifier_prefix, unqualified_name])
+
+
+# TODO big misnomer!
+# we already have a concept of Mathesar types (UI types) in the mathesar namespace.
+# maybe rename to just CustomType?
+# also, note that db layer should not be aware of Mathesar
+class MathesarCustomType(DatabaseType, Enum):
     """
     This is a list of custom Mathesar DB types.
-    Keys returned by get_available_types are of the format 'mathesar_types.VALUE'
     """
     EMAIL = 'email'
     MATHESAR_MONEY = 'mathesar_money'
     MULTICURRENCY_MONEY = 'multicurrency_money'
     URI = 'uri'
 
+    def __new__(cls, unqualified_id):
+        """
+        Prefixes a qualifier to this Enum's values.
+        `email` becomes something akin to `mathesar_types.email`.
+        """
+        qualified_id = get_qualified_name(unqualified_id)
+        instance = object.__new__(cls)
+        instance._value_ = qualified_id
+        return instance
+
+
+_non_canonical_alias_db_types = frozenset({
+    PostgresType.FLOAT,
+    PostgresType.TIME,
+    PostgresType.TIMESTAMP,
```

Comment:
I think so. I treated these a bit differently, because `DECIMAL` was being added by us, while these are there by default. I think I'll add a `TODO` comment about getting rid of them in a more holistic way.